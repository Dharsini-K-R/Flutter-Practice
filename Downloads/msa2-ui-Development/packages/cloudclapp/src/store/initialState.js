import { initialState as settings } from "./settings";
import { initialState as auth } from "./auth";
import { initialState as designations } from "./designations";

export default {
  auth,
  settings,
  designations,
};
