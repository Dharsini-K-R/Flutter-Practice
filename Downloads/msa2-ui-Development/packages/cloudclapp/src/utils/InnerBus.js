import EventEmitter from "events";

// List of all events available for usage with inner bus
const EVENTS = {
  LOGOUT: "LOGOUT",
  LOGIN: "LOGIN",
};

class InnerBus extends EventEmitter {
  evt = EVENTS;
}

export default new InnerBus();
