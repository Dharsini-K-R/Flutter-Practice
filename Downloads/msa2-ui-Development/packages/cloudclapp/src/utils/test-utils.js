import React from "react";
import { render } from "@testing-library/react";
import { renderHook, act } from "@testing-library/react-hooks";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { StylesProvider } from "@material-ui/core";
import { ThemeProvider } from "@material-ui/core";
import SnackbarProvider from "cloudclapp/src/SnackbarProvider";
import createTheme from "cloudclapp/src/styles/theme";
import thunk from "redux-thunk";
const middlewares = [thunk];

const msaTheme = createTheme();

// Make generated classNames consistent for snapshots
const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`;

const AllTheProviders = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={msaTheme}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </StylesProvider>
  );
};

const customRender = (component, options, initialRoute = "/") => {
  const componentWithRouter = (
    <Router basename={initialRoute}>{component}</Router>
  );

  return render(componentWithRouter, { wrapper: AllTheProviders, ...options });
};

const renderWithProvider = ({
  childComponent,
  mockStore,
  mockState = {},
  initialRoute,
}) => {
  const store = mockStore || configureMockStore(middlewares)(mockState);

  return {
    ...customRender(
      <Provider store={store}>{childComponent}</Provider>,
      null,
      initialRoute,
    ),
    store,
  };
};

const customRenderHook = (callback, options = {}) => {
  const wrapper = options.wrapper || AllTheProviders;

  return renderHook(callback, { ...options, wrapper });
};

// re-export everything
export * from "@testing-library/react";

// override render and renderHook methods
export {
  customRender as render,
  renderWithProvider,
  customRenderHook as renderHook,
  act as actOnHook,
  userEvent,
};
