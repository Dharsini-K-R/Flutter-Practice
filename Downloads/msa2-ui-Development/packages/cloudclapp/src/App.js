import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import { useSelector } from "react-redux";

import { getIsAuth } from "cloudclapp/src/store/auth";
import { getDesignationsReady } from "cloudclapp/src/store/designations";

import { ThemeProvider } from "@material-ui/core";

import AppWrapper from "cloudclapp/src/components/app/AppWrapper";
import createTheme from "cloudclapp/src/styles/theme";
import Login from "cloudclapp/src/routes/login/Login";
import MainMenu from "cloudclapp/src/components/menu/MainMenu";
import useTimeoutLogout from "cloudclapp/src/hooks/useTimeoutLogout";

function App() {
  const isAuth = useSelector(getIsAuth);
  const designationsReady = useSelector(getDesignationsReady);
  const state = useSelector((state) => ({
    // trim states to minimise rerender time
    auth: state.auth,
    designations: { permissions: state.designations.permissions },
  }));

  // This hook will logout the current user if he/she doesn't perform any click action on app
  // We can't invoke this from index.js because it should be wrapped under provider
  useTimeoutLogout(isAuth);

  if (!isAuth || !designationsReady) {
    const appTheme = createTheme(false);
    return (
      <ThemeProvider theme={appTheme}>
        <Route component={Login} />
      </ThemeProvider>
    );
  }

  const defaultRoute = "/dashboard";
  return (
    <AppWrapper>
      <Switch>
        <Route exact path="/">
          <Redirect to={defaultRoute} />
        </Route>
        <Route exact path="/sign-in">
          <Redirect to={defaultRoute} />
        </Route>
        {MainMenu.filter(({ getHidden }) =>
          getHidden ? !getHidden(state) : true,
        ).map(({ label, route, exact, component }) => (
          <Route key={label} path={route} exact={exact} component={component} />
        ))}
        <Redirect to={defaultRoute} />
      </Switch>
    </AppWrapper>
  );
}

export default App;
