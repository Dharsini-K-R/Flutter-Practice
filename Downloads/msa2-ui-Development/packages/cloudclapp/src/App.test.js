import React from "react";
import { render } from "cloudclapp/src/utils/test-utils";
import App from "./App";
import configureMockStore from "redux-mock-store";
import initialState from "cloudclapp/src/store/initialState";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
describe("App", () => {
  it("renders correctly", () => {
    const store = mockStore(initialState);
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
