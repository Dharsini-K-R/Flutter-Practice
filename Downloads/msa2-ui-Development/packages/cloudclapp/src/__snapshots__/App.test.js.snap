// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  class="MuiGrid-root makeStyles-loginWrapper MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
  >
    <div
      class="MuiGrid-root makeStyles-loginContainer MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root makeStyles-commonLoginFormWrapper MuiGrid-item"
      >
        <div
          class="MuiGrid-root makeStyles-loginFormSignIn MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiBox-root MuiBox-root makeStyles-logo"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                fill="none"
                focusable="false"
                style="fill: #DBE0ED; height: 50px; width: 300px;"
                viewBox="0 0 1457 240"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M56 136C56 91.8172 91.8172 56 136 56C153.572 56 169.821 61.6654 183.02 71.2694C170.511 43.4057 142.522 24 110 24C65.8172 24 30 59.8172 30 104C30 130.611 42.9926 154.187 62.9803 168.731C58.495 158.74 56 147.661 56 136Z"
                  fill="#46587E"
                />
                <path
                  d="M136 56C91.8172 56 56 91.8172 56 136C56 147.661 58.495 158.74 62.9803 168.731C76.179 178.335 92.4279 184 110 184C154.183 184 190 148.183 190 104C190 92.3388 187.505 81.2604 183.02 71.2694C169.821 61.6654 153.572 56 136 56Z"
                  fill="#7B93C1"
                />
                <path
                  clip-rule="evenodd"
                  d="M240 120C240 186.274 186.274 240 120 240C53.7258 240 0 186.274 0 120C0 53.7258 53.7258 0 120 0C186.274 0 240 53.7258 240 120ZM216 136C216 180.183 180.183 216 136 216C103.478 216 75.4892 196.594 62.9803 168.731C42.9926 154.187 30 130.611 30 104C30 59.8172 65.8172 24 110 24C142.522 24 170.511 43.4057 183.02 71.2694C203.007 85.8133 216 109.389 216 136Z"
                  fill="#34C186"
                  fill-rule="evenodd"
                />
                <path
                  d="M393.921 192.84C415.698 192.84 434.478 185.248 447.264 170.863L434.278 158.277C423.489 169.665 410.303 175.059 394.92 175.059C363.354 175.059 339.979 152.283 339.979 121.316C339.979 90.3494 363.354 67.5737 394.92 67.5737C410.303 67.5737 423.489 72.7682 434.278 84.156L447.264 71.5694C434.478 57.1848 415.698 49.7927 394.121 49.7927C351.566 49.7927 320 79.9605 320 121.316C320 162.672 351.566 192.84 393.921 192.84Z"
                  fill="#46587E"
                />
                <path
                  d="M468.972 191.241H488.151V43H468.972V191.241Z"
                  fill="#46587E"
                />
                <path
                  d="M566.127 192.44C597.893 192.44 621.069 169.864 621.069 138.298C621.069 106.732 597.893 84.3558 566.127 84.3558C534.361 84.3558 510.986 106.732 510.986 138.298C510.986 169.864 534.361 192.44 566.127 192.44ZM566.127 175.658C545.749 175.658 530.366 160.874 530.366 138.298C530.366 115.722 545.749 101.138 566.127 101.138C586.506 101.138 601.689 115.722 601.689 138.298C601.689 160.874 586.506 175.658 566.127 175.658Z"
                  fill="#46587E"
                />
                <path
                  d="M723.113 85.3547V140.096C723.113 162.872 710.326 175.259 690.348 175.259C672.167 175.259 661.778 164.87 661.778 144.092V85.3547H642.599V146.29C642.599 177.456 660.779 192.44 688.15 192.44C703.134 192.44 716.32 186.247 724.112 175.259V191.241H742.292V85.3547H723.113Z"
                  fill="#46587E"
                />
                <path
                  d="M855.523 43V101.338C846.732 89.9498 833.546 84.3558 818.562 84.3558C787.595 84.3558 765.019 105.933 765.019 138.298C765.019 170.663 787.595 192.44 818.562 192.44C834.145 192.44 847.731 186.447 856.322 174.459V191.241H874.702V43H855.523ZM820.16 175.658C799.782 175.658 784.399 160.874 784.399 138.298C784.399 115.722 799.782 101.138 820.16 101.138C840.339 101.138 855.722 115.722 855.722 138.298C855.722 160.874 840.339 175.658 820.16 175.658Z"
                  fill="#46587E"
                />
                <path
                  d="M953.146 192.44C972.126 192.44 987.909 184.449 996.3 169.665L981.716 160.474C974.923 170.863 964.534 175.658 952.946 175.658C932.169 175.658 916.585 161.274 916.585 138.298C916.585 115.722 932.169 101.138 952.946 101.138C964.534 101.138 974.923 105.933 981.716 116.322L996.3 106.932C987.909 92.1474 972.126 84.3558 953.146 84.3558C920.581 84.3558 897.206 106.732 897.206 138.298C897.206 169.864 920.581 192.44 953.146 192.44Z"
                  fill="#46587E"
                />
                <path
                  d="M1015.77 191.241H1034.95V43H1015.77V191.241Z"
                  fill="#46587E"
                />
                <path
                  d="M1142.32 82.9573V92.3472C1134.93 84.7553 1124.74 81.1592 1112.15 81.1592C1082.78 81.1592 1059.01 102.536 1059.01 136.9C1059.01 171.463 1082.78 193.04 1112.15 193.04C1126.34 193.04 1136.93 189.044 1144.12 180.453V191.241H1180.28V82.9573H1142.32ZM1120.14 162.872C1107.36 162.872 1097.37 153.482 1097.37 136.9C1097.37 120.517 1107.36 111.327 1120.14 111.327C1132.93 111.327 1142.92 120.517 1142.92 136.9C1142.92 153.482 1132.93 162.872 1120.14 162.872Z"
                  fill="#34C186"
                />
                <path
                  d="M1268.85 81.1592C1254.66 81.1592 1244.07 85.1549 1236.88 93.7457V82.9573H1200.72V230H1238.68V181.652C1246.07 189.244 1256.26 193.04 1268.85 193.04C1298.22 193.04 1321.99 171.463 1321.99 137.099C1321.99 102.736 1298.22 81.1592 1268.85 81.1592ZM1260.86 162.872C1248.07 162.872 1238.08 153.482 1238.08 137.099C1238.08 120.717 1248.07 111.327 1260.86 111.327C1273.64 111.327 1283.63 120.717 1283.63 137.099C1283.63 153.482 1273.64 162.872 1260.86 162.872Z"
                  fill="#34C186"
                />
                <path
                  d="M1403.77 81.1592C1389.58 81.1592 1378.99 85.1549 1371.8 93.7457V82.9573H1335.64V230H1373.6V181.652C1380.99 189.244 1391.18 193.04 1403.77 193.04C1433.13 193.04 1456.91 171.463 1456.91 137.099C1456.91 102.736 1433.13 81.1592 1403.77 81.1592ZM1395.77 162.872C1382.99 162.872 1373 153.482 1373 137.099C1373 120.717 1382.99 111.327 1395.77 111.327C1408.56 111.327 1418.55 120.717 1418.55 137.099C1418.55 153.482 1408.56 162.872 1395.77 162.872Z"
                  fill="#34C186"
                />
              </svg>
            </div>
          </div>
          <p
            class="MuiTypography-root makeStyles-subTitle MuiTypography-body1"
          >
            Member Login
          </p>
          <form
            class="makeStyles-form"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
            >
              <div
                class="MuiGrid-root makeStyles-formItem MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
              >
                <div
                  class="MuiFormControl-root"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                    style="width: 300px;"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
                      data-shrink="true"
                      for="SIGN_IN_EMAIL_INPUT"
                      id="SIGN_IN_EMAIL_INPUT-label"
                    >
                      Email Address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                        id="SIGN_IN_EMAIL_INPUT"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
                        >
                          <span>
                            Email Address
                             *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root makeStyles-formItem MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                    style="width: 300px;"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
                      data-shrink="true"
                      for="SIGN_IN_PASSWORD_INPUT"
                      id="SIGN_IN_PASSWORD_INPUT-label"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="on"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                        id="SIGN_IN_PASSWORD_INPUT"
                        required=""
                        type="password"
                        value=""
                      />
                      <div
                        class="MuiInputAdornment-root makeStyles-passwordToggle MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-marginDense"
                      >
                        <button
                          aria-label="Toggle password visibility"
                          class="MuiButtonBase-root MuiIconButton-root"
                          id="TOGGLE_SHOW_PASSWORD"
                          tabindex="-1"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <i
                              aria-hidden="true"
                              class="la la-eye"
                              role="presentation"
                              style="font-size: 20px;"
                            />
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
                        >
                          <span>
                            Password
                             *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button MuiButton-containedPrimary"
                  id="SIGN_IN_SUBMIT_BUTTON"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Login
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <span>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
                    id="FORGOT_PASSWORD_BTN_FORGOT_PASSWORD"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Forgot Password?
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </span>
              </div>
            </div>
          </form>
        </div>
        <div
          class="MuiGrid-root makeStyles-loginFormSignUp MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
        >
          <div
            class="MuiGrid-root makeStyles-formItem MuiGrid-item"
          >
            <p
              class="MuiTypography-root makeStyles-subTitle MuiTypography-body1"
            >
              Don't have an account?
            </p>
          </div>
          <div
            class="MuiGrid-root makeStyles-formItem MuiGrid-item"
          >
            <span
              class="MuiTypography-root makeStyles-signUpCaption MuiTypography-caption"
            >
              Click below to register an sign up today and get started!
            </span>
          </div>
          <div
            class="MuiGrid-root makeStyles-formItem MuiGrid-item"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button MuiButton-containedSecondary"
              href="/sign-up"
              id="SIGN_IN_GET_STARTED_BUTTON"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Get Started
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-copyrightSection MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
      >
        <img
          alt="UBIQUBE"
          class="makeStyles-companyLogo"
          src="companyLogo.png"
        />
        <span
          class="makeStyles-copyrightText"
        >
          Copyright 2022. All Rights Reserved. Ubiqube
        </span>
      </div>
    </div>
  </div>
</div>
`;
